modules/improvedigitalBidAdapter.js

```diff
@@ -192,539 +265,454 @@ export const spec = {
-            // Pop the current request and process it again in the next iteration
-            bidRequestObject.bid_request.imp.pop();
-            if (returnIdMappings) {
-              returnObject.idMappings.pop();
-            }
-            counter--;
-          }
-        }
+    // GPID
+    normalizedBidRequest.gpid = deepAccess(bid, 'ortb2Imp.ext.gpid') ||
+      deepAccess(bid, 'ortb2Imp.ext.data.pbadslot') ||
+      deepAccess(bid, 'ortb2Imp.ext.data.adserver.adslot');
 
-        if (writeLongRequest ||
-            !requestParameters.singleRequestMode ||
-            counter === impressionObjects.length - 1) {
-          returnObject.requests.push(this.formatRequest(requestParameters, bidRequestObject));
-          bidRequestObject = {
-            bid_request: this.createBasicBidRequestObject(requestParameters, extraRequestParameters)
-          };
-        }
-      }
-    }
+    // Interstitial Signal
+    normalizedBidRequest.interstitialSignal = deepAccess(bid, 'ortb2Imp.instl');
 
-    if (errors) {
-      returnObject.errors = errors;
-    }
+    return normalizedBidRequest;
+  },
 
-    return returnObject;
-  };
+  isValidSize(sizePair) {
+    return sizePair.length === 2 &&
+      isInteger(sizePair[0]) &&
+      isInteger(sizePair[1]) &&
+      sizePair[0] >= 0 &&
+      sizePair[1] >= 0;
+  }
+};
 
-  this.formatRequest = function(requestParameters, bidRequestObject) {
-    switch (requestParameters.returnObjType) {
-      case this.CONSTANTS.RETURN_OBJ_TYPE.URL_PARAMS_SPLIT:
-        return {
-          method: 'GET',
-          url: `${requestParameters.adServerBaseUrl}/${this.CONSTANTS.END_POINT}`,
-          data: `${this.CONSTANTS.AD_SERVER_URL_PARAM}${encodeURIComponent(JSON.stringify(bidRequestObject))}`
-        };
-      default:
-        const baseUrl = `${requestParameters.adServerBaseUrl}/` +
-          `${this.CONSTANTS.END_POINT}?${this.CONSTANTS.AD_SERVER_URL_PARAM}`;
-        return {
-          url: baseUrl + encodeURIComponent(JSON.stringify(bidRequestObject))
-        }
-    }
-  };
+const ID_REQUEST = {
+  buildImpression(bidRequest) {
+    const placementObject = ID_UTILITY.getNormalizedBidRequest(bidRequest);
+    const impressionObject = {
+      id: placementObject.id || getUniqueIdentifierStr(),
+      secure: window.location.protocol === 'https:' ? 1 : 0,
+    };
 
-  this.createBasicBidRequestObject = function(requestParameters, extraRequestParameters) {
-    let impressionBidRequestObject = {};
-    impressionBidRequestObject.secure = 1;
-    if (requestParameters.requestId) {
-      impressionBidRequestObject.id = requestParameters.requestId;
-    } else {
-      impressionBidRequestObject.id = getUniqueIdentifierStr();
-    }
-    if (requestParameters.domain) {
-      impressionBidRequestObject.domain = requestParameters.domain;
-    }
-    if (requestParameters.page) {
-      impressionBidRequestObject.page = requestParameters.page;
+    if (placementObject.bidFloor) {
```

Comment:
it might be a good idea to create a utility function doing:
```
if (value) {
    deepSetValue(obj, <path>, value)
}
```
This would reduce the 3 line deepSetValue commands in 1 line.